### ISSUE TITLE
[idea] New startup wizard

### ISSUE THREAD

**Issue by toby63** [mumble-voip/mumble#4137]
**Context:**
First-Start-Configuration and Information

**Description:**
This is intended to be the central issue for an idea that @Krzmbrzl mentioned several times.
Right now there is only the standard Audio Wizard and it might be useful to have an additional extended startup wizard (that also includes the Audio Wizard).

**Potencial Contents:**
* Audio Wizard 
    * [new] Echo cancellation (see also: #4125 #4127 )
* In-game-overlay (#3515 )
* anonymous statistics (to seperate it from the audio wizard)
* certificate creation (already implemented)
* public server list (Visibility and Activation) (#4070)
* update notification
* UI-options:
    * theme 
    * (once implemented) font size (#1850) )
    * (if implemented) symbole size and position
    * (if implemented) channel/user-flags (#1632) 

* potencial features:
    * E2E-encryption (#1813)

**Server management (maybe seperate wizard for that):**
* ACLs (easy or complex) (#3247)

**Comment by Krzmbrzl** [comment]
ACLs are server-specific. What do you want to configure in a startup wizard?

**Comment by toby63** [comment]
@Krzmbrzl 
> ACLs are server-specific. What do you want to configure in a startup wizard?

Thats why its a "potencial feature", it is related to my idea from #3247 , to add an easier and a more complex version of ACLs (aka one without and one with inheritage).
So in the startup wizard the user can choose which one they want to have.
It is only a placeholder, but if you want to delete it... or we could strike it out for now.

**Comment by Krzmbrzl** [comment]
Ah okay I see. Yeah once something like this is implemented for the ACL (if it is), then that'd indeed be a good candidate for the startup wizard.

For now I'd only focus on stuff that already exists though :point_up:

**Comment by deluxghost** [comment]
Will there be any "easy mode" for new users? Compare to other apps, mumble is more difficult to getting started.

Some players are thinking "wtf is a cert i dont wanna click it" and close all windows to feel safer.

About the audio wizard, it could just make sure the user can talk and listen (in one or two steps of the wizard), he doesn't need to care what is position audio or blablabla.

**Comment by streaps** [comment]
Important settings, but I don't like wizards in the sense of clicking through a lot of options and dialogs. Multiple "wizards" are even worse.

My problem with wizards is that I have to decide about things I don't fully understand yet. I also have to places and ways for changing preferences.

Plumble / Mumla works fine without any wizard. Use good default settings and improve the Settings UI. I think all these points need different solutions and not a new startup wizard.

**Comment by toby63** [comment]
@streaps 
While it is of course a valid point to minimize or even leave out a startup-wizard, I think this is more a "religious" debate.

An Example:
I remember when a friend wondered why he had installed so many annoying new programs.
It turned out that he installed one of those fancy free programs from whatever site and he did not even care to look at the steps during installation, so he "unintentionally" installed all programs that were offered during the installation.

So this tells us, there are at least two types of users:
1. the impatient who never want to read anything.
2. the patient persons who read through things.

For group 1 startup wizards are horrible.
For group 2 startup wizards are very useful.

> Multiple "wizards" are even worse.

To clarify, there shall not be multiple startup wizards.
There will only be multiple wizards, so that users can use the Audio- and Certificate-Wizard again later (these could be renamed though, to clarify the purpose).

> My problem with wizards is that I have to decide about things I don't fully understand yet.

Well a good wizard explains things good enough.

> Plumble / Mumla works fine without any wizard. Use good default settings and improve the Settings UI. 

Well I don't know about Plumble, but the point is (and that was @Krzmbrzl's intention of this idea (i guess)) to inform the users, that certain options even exist.

@deluxghost 
> Some players are thinking "wtf is a cert i dont wanna click it" and close all windows to feel safer.

Well then thats a lack of description or a lack of basic understanding by the user.
Nonetheless it is an option to just create it, without asking the user (I guess many programs do it this way, because many programs also need encryption).

> About the audio wizard, it could just make sure the user can talk and listen (in one or two steps of the wizard), he doesn't need to care what is position audio or blablabla.

Well that is not so easy, the user needs to adjust things (while it would in theory maybe be possible to set this automatic, that is complicated to implement).

tldr:
I guess some things are necessary in a startup wizard (though of course I also know people who ignored it (even though they were completely clueless about what to do)).
But the scope can of course be discussed and the idea of @deluxghost for an easy and an advanced wizard is good.

**Comment by streaps** [comment]
>  While it is of course a valid point to minimize or even leave out a startup-wizard, I think this is more a "religious" debate.

is it? so you are a strong believer in startup wizards?

Why? Is there any clear evidence that it would improve usability for Mumle?

> So this tells us, there are at least two types of users:
> 
>     the impatient who never want to read anything.
>     the patient persons who read through things.
> 
> For group 1 startup wizards are horrible.
> For group 2 startup wizards are very useful.

I don't believe this. Do you have any real evidence (and not just anecdotal) that this is a valid assumption?

For me it reads like:

- We have to improve stuff for new users
- Wizards!

1. Collect a lot of potential settings
2. ?
3. Profit

The two Wizards we have now are bad UX. I would rather to get rid of them or simplify instead of adding additional layers of wizardry.

What about trying to improve the most annoying UI in Mumble first: the audio settings + audio tuning wizard. This thing is insane. It wants me to deinstall Mumble after the first invocation.

**Comment by toby63** [comment]
@streaps 
> Is there any clear evidence that it would improve usability for Mum[b]le?

I never said it would.
I see it like this:
1. The wizard is in fact optional (afaik) and will surely stay this way, so everyone of the impatient can just close it and has to accept needing to adjust settings later.
2. Thats exactly the point of a wizard, you can show users the most important (maybe even necessary (like audio adjustments)) settings and explain them on the way.
3. I think wizards are a good thing and far better than annoying pop-ups that many programs use to explain functions.

> Do you have any real evidence (and not just anecdotal) that this is a valid assumption?

You can search for marketing studies if you want, to prove me that wizards are nonsense that no one wants.

> What about trying to improve the most annoying UI in Mumble first: the audio settings + audio tuning wizard. This thing is insane. It wants me to deinstall Mumble after the first invocation.

I would like to know what is bad about the UI?
I don't see why and how the UI is bad and how especially the audio settings can improve.
Could you describe that in more detail?

**Comment by streaps** [comment]
I have my doubts that marketing studies will teach you good interface design, but the market for Mumble doesn't look good ;)

https://trends.google.com/trends/explore?cat=13&date=2012-01-01%202020-02-01&q=%2Fm%2F02x7h_d,%2Fg%2F11cjpb9d52,%2Fm%2F03z9t7,%2Fm%2F011c8m4f

There is so much wrong with the audio wizard and the UI, that it takes some work to write a useful reply. I'll try to write a proposal for some improvements, if I find the time.

Don't get me wrong, I like Mumble and I don't know any other open source software that has all the important features. It's just that it's very hard to convince an average user to use Mumble, when every other software is much easier to understand.

**Comment by Krzmbrzl** [comment]
> I'll try to write a proposal for some improvements, if I find the time.

That'd be great. I think It'll probably best to open a separate issue for that though :point_up: 
I never understood what's hard to get in Mumble (at least for getting it to work for the basic stuff), but then again I'm not really an "average" user. 
Thus it might really be valuable to get pointed to problematic areas :)

**Comment by Krzmbrzl** [comment]
If implemented the wizard might include some description about the audio filter options (see #4127 )
